lib/newrelic_httprb/wrappers.rb

```diff
@@ -0,0 +1,51 @@
+module NewRelicHTTP
+  class HTTPResponse
+    attr_reader :response
+
+    def initialize(response)
+      @response = response
+    end
+
+    def [](key)
+      response.headers.each do |k,v|
+        if key.downcase == k.downcase
+          return v
+        end
+      end
+      nil
+    end
+
+    def to_hash
+      response.headers
+    end
+  end
```

Comment:
Using the class equality was breaking the tests in the NewRelic RPM test suite for a reason I wasn't able to track. Due to a slightly different behaviour of `HTTP` response class being wrapped here, this is now slightly different from `HTTPClientResponse` so I think no copyright issues will arise.